### PR TITLE
Tsdb backend v2

### PR DESCRIPTION
**Pull request by talIguaz**

### CODE REVIEW

File: backends/tsdb/reader.go [hunk 1]
```diff
@@ -63,18 +74,64 @@ func (b *Backend) Read(request *frames.ReadRequest) (frames.FrameIterator, error
...
-	qry, err := adapter.Querier(nil, from, to)
+	qry, err := adapter.QuerierV2()
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to initialize Querier")
 	}
 
 	iter := tsdbIterator{request: request}
 	name := ""
+	if len(request.Columns) > 0 {
+		name = request.Columns[0]
```
Comment (tebeka) [inline backends/tsdb/reader.go]: Use [strings.Join](https://golang.org/pkg/strings/#Join)

File: builder.go [hunk 2]
```diff
@@ -42,13 +44,15 @@ func NewSliceColumnBuilder(name string, dtype DType, size int) ColumnBuilder {
 		Dtype: pb.DType(dtype),
 		Size:  int64(size),
 	}
+
 	// TODO: pre alloate array. Note that for strings we probably don't want to
 	// do this since we'll allocate strings twice - zero value then real value
-	return &sliceColumBuilder{msg: msg}
+	return &sliceColumBuilder{msg: msg, originalSize: int64(size)}
```
Comment (tebeka) [inline builder.go]: Use `NewSliceColumBuilder`, resist the temptation to use internal structures as much as posslbie :)

File: builder.go [hunk 3]
```diff
@@ -193,9 +199,62 @@ func resizeInt64(buf []int64, size int) []int64 {
...
+		return
+	}
+
+	b.resize(int(b.originalSize))
+	b.fillDefaultValues(currentSize, int(b.originalSize))
+}
+
+func (b *sliceColumBuilder) fillDefaultValues(from, to int) {
+	switch b.msg.Dtype {
+	//case pb.DType_INTEGER:
```
Comment (tebeka) [inline builder.go]: Delete commented code

File: builder.go (hunk 2 above)
Comment (talIguaz) [reply builder.go]: this code is inside the NewSliceColumBuilder 😅

**Pull request merged**